client/sites-dashboard/components/sites-table-row.tsx

```diff
@@ -142,7 +147,23 @@ export default function SitesTableRow( { site }: SiteTableRowProps ) {
 							href={ getDashboardUrl( site.slug ) }
 							title={ __( 'Visit Dashboard' ) }
 						>
-							<SiteIcon siteId={ site.ID } size={ 50 } />
+							<SiteThumbnail
+								mShotsUrl={ shouldUseScreenshot ? site.URL : undefined }
+								alt={ site.name }
```

Comment:
For more semantic correctness, this `alt` probably should be a localized version of 'Screenshot of %s' or similar.

Reply:
You could be right, but I'm planning to put together a PR that makes sure the accessible labels for these links are working nicely. I'll read up what the best practices are, but I suspect that it's going to be more important to correctly label the link than to label the iamge.